src/tarball.py

```diff
@@ -98,27 +98,30 @@ def _push_tarball(self, config, describe):
         os.unlink(tarball_path)
         return tarball
 
-    def _create_tarball_node(self, checkout_node, status, result):
-        node = {
+    def _create_tarball_node(self, checkout_node):
+        tarball_node = {
             'parent': checkout_node['_id'],
             'name': 'tarball',
             'path': checkout_node['path'] + ['tarball'],
-            'artifacts': checkout_node['artifacts'],
             'revision': checkout_node['revision'],
-            'status': status,
-            'result': result,
         }
-        self._db.submit({'node': node})
+        return self._db.submit({'node': tarball_node})[0]
+
+    def _make_tarball_node_available(self, tarball_node, tarball):
+        tarball_node.update({
+            'state': 'available',
+            'artifacts': {
+                'tarball': urllib.parse.urljoin(self._storage_url, tarball),
+            },
+        })
+        return self._db.submit({'node': tarball_node})
```

Comment:
The return value is not being used.
```suggestion
        self._db.submit({'node': tarball_node})
```